packages/asset-swapper/src/utils/market_operation_utils/index.ts

```diff
@@ -78,19 +93,16 @@ export class MarketOperationUtils {
             MarketOperation.Sell,
         );
 
-        const prunedNativePath = pruneDustFillsFromNativePath(
-            createSellPathFromNativeOrders(nativeOrdersWithFillableAmounts),
-            takerAmount,
-            _opts.dustFractionThreshold,
-        );
-        const clippedNativePath = clipPathToInput(sortFillsByPrice(prunedNativePath), takerAmount);
```

Comment:
Clipping native orders is dumb and inaccurate because a native order can only be clipped at fill time if it's actually positioned last in the order queue.